### PR TITLE
Add recipes for dmri-amico and dmri-dicelib

### PR DESCRIPTION
**Pull request by pauldmccarthy**
Howdy, this PR adds two recipes - one for `dmri-amico`, along with a core library that it depends on - `dmri-dicelib`:

 - https://github.com/daducci/AMICO/
 - https://pypi.org/project/dmri-dicelib/
 - https://pypi.org/project/dmri-amico/

AMICO is a popular diffusion MRI microstructure modelling framework.

@daducci and @nightwnvol - would you be able to leave a comment here, confirming that you are happy to be added as package maintainers? Thanks!

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by nightwnvol** [comment]
Happy to be a maintainer! Thanks @pauldmccarthy

File: recipes/dmri-amico/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{% set name = "dmri-amico" %}
+{% set version = "2.0.1" %}
+{% set sha256 = "08fe9fe14554656fcfe9862dcdbb40f5cf9fc1cb9af5685435190110b83f1902" %}
```
Comment (ocefpaf) [inline recipes/dmri-amico/meta.yaml:3]:
```suggestion
```

File: recipes/dmri-amico/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/dmri-amico/meta.yaml:10]:
```suggestion
```

File: recipes/dmri-amico/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/dmri-amico/meta.yaml:12]:
```suggestion
  sha256: 08fe9fe14554656fcfe9862dcdbb40f5cf9fc1cb9af5685435190110b83f1902
```

File: recipes/dmri-dicelib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{% set name = "dmri-dicelib" %}
+{% set version = "1.0.3" %}
+{% set sha256 = "5a2e315e2bebed0cbf19e300a2328827903d77086824ae8f04b0410f65585cc2" %}
```
Comment (ocefpaf) [inline recipes/dmri-dicelib/meta.yaml:3]:
```suggestion
```

File: recipes/dmri-dicelib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/dmri-dicelib/meta.yaml:10]:
```suggestion
```

File: recipes/dmri-dicelib/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/dmri-dicelib/meta.yaml:12]:
```suggestion
  sha256: 5a2e315e2bebed0cbf19e300a2328827903d77086824ae8f04b0410f65585cc2
```

File: recipes/dmri-amico/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  patches:
+    - setup.py.patch
```
Comment (ocefpaf) [inline recipes/dmri-amico/meta.yaml:14]: Can you add a comment on why this patch is needed and/or if it was upstreamed?

Comment (pauldmccarthy) [reply recipes/dmri-amico/meta.yaml:17]: @ocefpaf Sure - there is an [open PR](https://github.com/daducci/AMICO/pull/179) upstream regarding this, and I've added a comment in `meta.yaml` explaining why it is needed for this version.

**Comment by pauldmccarthy** [comment]
Hi @conda-forge/help-python, just a gentle reminder that this recipe is ready for [re-]review - thanks!

**Comment by pauldmccarthy** [comment]
Hi @conda-forge/help-python, another poke to try and get this recipe moving - thanks!

**Pull request merged**